cmd/legacy_param_mapper_test.go

```diff
@@ -0,0 +1,347 @@
+						BackupFileCount: 2,
+						Compress:        true,
+					},
+				},
+				FileCache: cfg.FileCacheConfig{
+					MaxSizeMb:                20,
+					CacheFileForRangeRead:    true,
+					EnableParallelDownloads:  true,
+					ParallelDownloadsPerFile: 3,
+					MaxParallelDownloads:     6,
+					DownloadChunkSizeMb:      9,
+					EnableCrc:                true,
+				},
+				CacheDir: cfg.ResolvedPath(path.Join(os.Getenv("HOME"), "cache-dir")),
+				MetadataCache: cfg.MetadataCacheConfig{
+					TtlSecs:            200,
+					TypeCacheMaxSizeMb: 7,
+					StatCacheMaxSizeMb: 4,
+				},
+				List: cfg.ListConfig{
+					EnableEmptyManagedFolders: true,
+					KernelListCacheTtlSecs:    30,
+				},
+				GcsConnection: cfg.GcsConnectionConfig{
+					GrpcConnPoolSize: 29,
+					ClientProtocol:   cfg.Protocol("grpc")},
+				GcsAuth:   cfg.GcsAuthConfig{AnonymousAccess: true},
+				EnableHns: true,
+				FileSystem: cfg.FileSystemConfig{
+					DisableParallelDirops: true,
+					IgnoreInterrupts:      true,
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			testName: "Test overlapping flags and configs set.",
+			legacyFlagStorage: &flagStorage{
+				LogFile:                   "~/Documents/log-flag.txt",
+				LogFormat:                 "json",
+				IgnoreInterrupts:          false,
+				AnonymousAccess:           false,
+				KernelListCacheTtlSeconds: -1,
+				ClientProtocol:            mountpkg.HTTP2,
+			},
+			mockCLICtx: &mockCLIContext{
+				isFlagSet: map[string]bool{
+					"log-file":                   true,
+					"log-format":                 true,
+					"ignore-interrupts":          true,
+					"anonymous-access":           true,
+					"kernel-list-cache-ttl-secs": true,
+				},
+			},
+			legacyMountConfig: &config.MountConfig{
+				LogConfig: config.LogConfig{
+					FilePath: "~/Documents/log-config.txt",
+					Format:   "text",
+					Severity: "INFO",
+				},
+				FileSystemConfig: config.FileSystemConfig{
+					IgnoreInterrupts: true,
+				},
+				GCSAuth: config.GCSAuth{
+					AnonymousAccess: true,
+				},
+				ListConfig: config.ListConfig{
+					KernelListCacheTtlSeconds: 100,
+				},
+			},
+			expectedConfig: &cfg.Config{
+				Logging: cfg.LoggingConfig{
+					FilePath: cfg.ResolvedPath(path.Join(os.Getenv("HOME"), "/Documents/log-flag.txt")),
+					Format:   "json",
+					Severity: "INFO",
+				},
+				FileSystem: cfg.FileSystemConfig{
+					IgnoreInterrupts: false,
+				},
+				GcsAuth: cfg.GcsAuthConfig{
+					AnonymousAccess: false,
+				},
+				List: cfg.ListConfig{
+					KernelListCacheTtlSecs: -1,
+				},
+				GcsConnection: cfg.GcsConnectionConfig{
+					ClientProtocol: cfg.Protocol("http2"),
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, tc := range populateConfigFromLegacyFlags {
+		t.Run(tc.testName, func(t *testing.T) {
+			resolvedConfig, err := PopulateNewConfigFromLegacyFlagsAndConfig(tc.mockCLICtx, tc.legacyFlagStorage, tc.legacyMountConfig)
+
+			require.Equal(t, tc.expectedErr, err)
```

Comment:
I think the usual way is to
if assert.Nil(...) {
  assert.Equal(..)
}

Curious why you have adopted this style.